### PR TITLE
fix: gas sensor automatic record without sensor

### LINKED ISSUE #2093: bug: Recording start even when device not connected
**Actual Behaviour**
Recording can be started even if the device is not connected to mobile or mobile does not have inbuilt feature support

**Expected Behaviour**
If any device is not connected then for that device recording should not start

**Steps to reproduce it**
Open any device which is not connected or whose sensor are not present in mobile(Gas Sensor)
Click on Record button
See the error

**Screenshots of the issue**
recording_issue

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by SpandanBasu**
Fixes #2093 

**Changes**: Changes are made in PSLabSensor.java at onOptionsItemSelected(MenuItem item) function a check is added for gas sensor. (If a gas sensor is connected, only then a recording of data will start)

**Screenshot/s for the changes**: 
https://user-images.githubusercontent.com/62503399/103482147-193a8280-4e05-11eb-829d-38a307c52344.mp4
https://user-images.githubusercontent.com/62503399/103482149-1a6baf80-4e05-11eb-9afc-de0b85f6525b.mp4
https://user-images.githubusercontent.com/62503399/103482150-1b044600-4e05-11eb-9a74-d8601a9d366f.mp4

**Checklist**: 
- [x ] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x ] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**: 
[app-fdroid-debug.zip](https://github.com/fossasia/pslab-android/files/5761787/app-fdroid-debug.zip)

### CODE REVIEW

File: app/src/main/java/io/pslab/models/PSLabSensor.java [hunk 1]
```diff
@@ -334,6 +343,41 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+                    builder1.setNegativeButton(
+                            R.string.cancel,
+                            new DialogInterface.OnClickListener() {
+                                public void onClick(DialogInterface dialog, int id) {
+                                    dialog.cancel();
+                                }
+                            });
+                    builder1.setIcon(android.R.drawable.ic_dialog_alert);
+                    AlertDialog alert11 = builder1.create();
+                    alert11.show();
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/models/PSLabSensor.java:374]: No need of many dialog boxes

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/pslab/models/PSLabSensor.java (hunk 1 above)
Comment (SpandanBasu) [reply app/src/main/java/io/pslab/models/PSLabSensor.java:374]: ok, sure I will resolve these issues very soon. Thanks.

**Comment by SpandanBasu** [comment]
All changes are done, Please review this

File: app/src/main/java/io/pslab/models/PSLabSensor.java [hunk 2]
```diff
@@ -334,6 +342,18 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+            {
+                try {
+                    CustomSnackBar.showSnackBar(sensorParentView, NOT_CONNECTED, null, null, Snackbar.LENGTH_LONG);
+
+                }catch (IllegalStateException e)
+                {
+                    e.printStackTrace();
+                }
+            }
+
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/models/PSLabSensor.java:356]: Reformat this file using Android Studio

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
Looks good. Need to reformat to fix spacing and indentation

File: app/src/main/java/io/pslab/models/PSLabSensor.java (hunk 2 above)
Comment (SpandanBasu) [reply app/src/main/java/io/pslab/models/PSLabSensor.java:356]: Sure

**Comment by SpandanBasu** [comment]
Spaces and indentation perfected

**Pull request closed without merging**